easybuild/tools/utilities.py

```diff
@@ -223,6 +223,25 @@ def nub(list_):
     return [x for x in list_ if x not in seen and not seen_add(x)]
 
 
+def unique_ordered_append(base, extra):
+    """Append elements of extra list to another base list keeping order and without duplicates"""
+    if extra and isinstance(extra, str):
```

Comment:
I'm still not sure supporting a string argument is correct and omitting the empty string. It is an undocumented feature now and e.g. `unique_ordered_append(base, 'foo') == unique_ordered_append(base, ['foo'])` but `unique_ordered_append(base, '') != unique_ordered_append(base, [''])` which could be unexpected. Hence I'd just disallow strings to avoid the ambiguity

Reply:
ok, I see your point. I forbid strings altogether and changed the name of the method to `unique_ordered_extend` to remove any expectation of it working like `List.append()`.